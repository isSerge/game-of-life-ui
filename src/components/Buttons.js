import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
    margin: 5px 10px;
    padding: 10px 20px;
    border-radius: 15px;
    text-transform: uppercase;
    outline: none;

    &:hover {
        background-color: ${({ theme }) => theme.selectedColor};
    }
`

const Wrapper = styled.div`
    margin: 20px 0;
`

const Buttons = ({ startTicks, nextTick, pauseTicks, refreshTicks }) => (
    <Wrapper>
        <Button onClick={startTicks}>start</Button>
        <Button onClick={nextTick}>next</Button>
        <Button onClick={pauseTicks}>pause</Button>
        <Button onClick={refreshTicks}>refresh</Button>
    </Wrapper>
)

export default Buttons
