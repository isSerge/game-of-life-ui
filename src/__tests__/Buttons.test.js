import React from 'react'
import Buttons from '../components/Buttons'
import renderer from 'react-test-renderer'

test('Buttons matches snapshot', () => {
    const props = {
        startTicks: jest.fn(),
        nextTick: jest.fn(),
        pauseTicks: jest.fn(),
        refreshTicks: jest.fn(),
    }
    const component = renderer.create(<Buttons {...props} />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})
