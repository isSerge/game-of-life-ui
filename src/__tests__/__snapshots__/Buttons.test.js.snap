// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons matches snapshot 1`] = `
<div
  className="sc-bwzfXH hDHdtQ"
>
  <button
    className="sc-bdVaJa btTGJZ"
    onClick={[MockFunction]}
  >
    start
  </button>
  <button
    className="sc-bdVaJa btTGJZ"
    onClick={[MockFunction]}
  >
    next
  </button>
  <button
    className="sc-bdVaJa btTGJZ"
    onClick={[MockFunction]}
  >
    pause
  </button>
  <button
    className="sc-bdVaJa btTGJZ"
    onClick={[MockFunction]}
  >
    refresh
  </button>
</div>
`;
